engine/src/main/java/com/datatorrent/stram/util/LoggerUtil.java

```diff
@@ -464,4 +470,68 @@ public static boolean removeAppender(String name)
     }
     return names;
   }
+
+  /**
+   * Makes MDC properties
+   */
+  private static void makeMDCProperties()
+  {
+    if (setMDCProperties) {
+
+      MDC.put("node", getCurrentHost());
+
+      String value = System.getenv("HADOOP_USER_NAME");
+      if (value != null) {
+        MDC.put("user", value);
+      }
+
+      value = System.getenv("CONTAINER_ID");
+      if (value != null) {
+        MDC.put("containerId", value);
+      }
+
+      Properties properties = System.getProperties();
+      String longName = LogicalPlan.APPLICATION_NAME.getLongName();
+      if (properties.contains(longName)) {
+        MDC.put("application", properties.getProperty(longName));
+      }
+
+      if (MDC.get("service") == null) {
+        MDC.put("service", getService());
+      }
+
+      setMDCProperties = false;
+    }
+  }
+
+  /**
+   * Returns the service name
+   * @return Service name
+   */
+  private static String getService()
+  {
+    StackTraceElement[] stack = Thread.currentThread().getStackTrace();
+    String mainClass = stack[stack.length - 1].getClassName();
+    if (mainClass.equals("com.datatorrent.stram.cli.ApexCli")) {
+      return "apexclient";
+    } else if (mainClass.equals("com.datatorrent.stram.StreamingAppMaster")) {
+      return "apexmaster";
+    } else if (mainClass.equals("com.datatorrent.stram.engine.StreamingContainer")) {
+      return "apexcontainer";
+    }
+    return "unknown";
+  }
+
+  /**
+   * Returns the current host name
+   * @return The current host name
+   */
+  private static String getCurrentHost()
```

Comment:
Does not belong to this class. In case such helper method does not exist already, it should be added to one of the common utils classes.